src/lib.rs

```diff
@@ -136,27 +139,23 @@ impl State {
 }
 
 /// Shows the docking hierarchy inside a `Ui`.
-pub fn show<Ctx>(
-    ui: &mut Ui,
-    id: Id,
-    style: &Style,
-    tree: &mut Tree<Ctx>,
-    context: &mut Ctx,
-) {
+pub fn show<Ctx>(ui: &mut Ui, id: Id, style: &Style, tree: &mut Tree<Ctx>, context: &mut Ctx) {
     let mut state = State::load(ui.ctx(), id);
-
     let mut rect = ui.max_rect();
-    if tree.is_empty() || tree[NodeIndex::root()].is_none() {
-        //ui.painter().rect_filled(rect, 0.0, style.background);
-        // TODO: splash screen here?
-        return;
-    }
 
-    let rect = {
-        rect.min += style.padding.left_top() + style.padding.left_top();
-        rect.max -= style.padding.right_bottom() + style.padding.right_bottom();
-        rect
-    };
+    if let Some(margin) = style.padding {
+        rect = {
+            rect.min += margin.left_top();
+            rect.max -= margin.right_bottom();
+            ui.painter().rect(
+                rect,
+                margin.top,
+                style.separator_color,
+                Stroke::new(margin.top, style.border_color),
+            );
+            rect
+        };
+    }
```

Comment:
This can be reduced to:
```rust
if let Some(margin) = style.padding {
    rect.min += margin.left_top();
    rect.max -= margin.right_bottom();
    ui.painter().rect(
        rect,
        margin.top,
        style.separator_color,
        Stroke::new(margin.top, style.border_color),
    );
}
```